runtime/near-vm-runner/src/logic/logic.rs

```diff
@@ -1774,9 +1774,18 @@ impl<'a> VMLogic<'a> {
         let (receipt_idx, sir) = self.promise_idx_to_receipt_idx_with_sir(promise_idx)?;
         let receiver_id = self.ext.get_receipt_receiver(receipt_idx);
         let is_receiver_implicit =
-            self.config.implicit_account_creation && receiver_id.is_implicit();
-        let send_fee = transfer_send_fee(self.fees_config, sir, is_receiver_implicit);
-        let exec_fee = transfer_exec_fee(self.fees_config, is_receiver_implicit);
+            self.config.implicit_account_creation && receiver_id.get_account_type().is_implicit();
+        let send_fee = transfer_send_fee(
+            self.fees_config,
+            sir,
+            is_receiver_implicit,
+            receiver_id.get_account_type(),
```

Comment:
Can you get rid of the is_receiver_implicit now that the account type is provided? It's seems duplicate information. Also it's hard to tell from here if near and eth implicit should be treated the same, only passing the account type will make it easier to make that determination where it's meaningful.

Reply:
Actually it won't quite work because of the config, but perhaps it could work if you also provide the self.config.implicit_account_creation as an argument?